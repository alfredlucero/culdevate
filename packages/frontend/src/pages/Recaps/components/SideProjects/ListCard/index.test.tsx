import React from "react";
import { render } from "@testing-library/react";
import SideProjectsListCard from "./index";

describe("<SideProjectsListCard />", () => {
  test("should render without error", () => {
    const { container } = render(
      <SideProjectsListCard
        onClickAdd={() => {}}
        onClickView={() => {}}
        count={10}
        className="extra-list-class"
        testId="testId"
      />,
    );

    expect(container).toMatchSnapshot();
  });
});
