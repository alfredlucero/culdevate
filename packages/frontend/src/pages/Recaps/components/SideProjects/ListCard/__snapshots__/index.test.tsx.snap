// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SideProjectsListCard /> should render without error 1`] = `
<div>
  <div
    class="bg-white shadow-md rounded px-8 py-6 w-64 extra-list-class"
    data-testid="testId"
  >
    <div
      class="mb-6 flex justify-center"
    >
      <div
        class="flex justify-center items-center p-3 rounded-lg bg-gray-200 recap-icon"
      >
        <div
          class="font-sans text-3xl text-teal-400"
          id=""
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-project-diagram fa-w-20 "
            data-icon="project-diagram"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 640 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M384 320H256c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V352c0-17.67-14.33-32-32-32zM192 32c0-17.67-14.33-32-32-32H32C14.33 0 0 14.33 0 32v128c0 17.67 14.33 32 32 32h95.72l73.16 128.04C211.98 300.98 232.4 288 256 288h.28L192 175.51V128h224V64H192V32zM608 0H480c-17.67 0-32 14.33-32 32v128c0 17.67 14.33 32 32 32h128c17.67 0 32-14.33 32-32V32c0-17.67-14.33-32-32-32z"
              fill="currentColor"
            />
          </svg>
        </div>
      </div>
    </div>
    <h4
      class="font-mono font-bold text-gray-800 text-xl mb-4 text-center"
    >
      Side Projects
    </h4>
    <div
      class="flex justify-center mb-4"
    >
      <button
        class="font-bold py-2 px-4 rounded bg-transparent text-teal-700 border border-teal-500 hover:bg-teal-500 hover:text-white hover:border-transparent mr-2"
        type="button"
      >
        <div
          class="flex items-center"
        >
          View
        </div>
      </button>
      <button
        class="font-bold py-2 px-4 rounded font-bold py-2 px-4 rounded bg-teal-500 text-white hover:bg-teal-700 "
        type="button"
      >
        <div
          class="flex items-center"
        >
          Add
        </div>
      </button>
    </div>
    <div
      class="flex justify-end"
    >
      <div
        class="rounded bg-gray-200 text-gray-700 text-sm p-2 w-8 flex justify-center items-center"
      >
        10
      </div>
    </div>
  </div>
</div>
`;
